### ISSUE TITLE
Remove woven from next release of twisted.web

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11060]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#1522|
|Type|task|
|Created|2006-03-01 16:38:51Z|
|Branch|https://github.com/twisted/twisted/tree/remove-woven-1522-2|

Enough is enough.

<details><summary>Searchable metadata</summary>

```
trac-id__1522 1522
type__task task
reporter__itamarst itamarst
priority__highest highest
milestone__Twisted_9_0 Twisted-9.0
branch__branches_remove_woven_1522_2 branches/remove-woven-1522-2
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1141231131000000 1141231131000000
changetime__1231520519000000 1231520519000000
version__ 
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst set owner to foom|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

docs shouldn't be removed until the code is

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight set owner to @jyknight|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|collab commented|
|-|-|

Replying to [exarkun](#note_2):
> docs shouldn't be removed until the code is

Where can that code be found?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

`twisted.web.woven` :)

therve is working on a branch that removes the use of it from `twisted.web.static`.  Not sure if there are any other uses in Twisted.

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|collab set owner to collab<br>collab set status to `assigned`|
|-|-|

Replying to [exarkun](#note_5):
> `twisted.web.woven` :)
>
> therve is working on a branch that removes the use of it from `twisted.web.static`.  Not sure if there are any other uses in Twisted.

Allright. Also saw pylint mentioning a cyclic import in the woven package.. good to hear its going ;) I could make a patch that scrapes out the documentation for that branch, which one is that?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra set owner to @therve|
|-|-|

assigning it to therve cause he's working on the woven cleanup.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

Also to remove:

  * twisted/web/widgets.py
  * the woven support in twisted/web/tap.py
  * twisted/web/woven/ and associated tests

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

(In [25874]) Branching to 'remove-woven-1522'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

Splat.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_13"></a>|@therve set owner to @exarkun|
|-|-|

Awesome, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun commented|
|-|-|

(In [25918]) Branching to 'remove-woven-1522-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_15"></a>|@exarkun set status to `closed`|
|-|-|

(In [25924]) Merge remove-woven-1522-2

Author: exarkun
Reviewer: therve
Fixes: [#11060](https://github.com/twisted/twisted/issues/11060)

Remove `twisted.web.woven` and all remaining code which used anything from
that package as well as the documentation for Woven.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
